app/controllers/admin/publishers_controller.rb

```diff
@@ -2,14 +2,43 @@ class Admin::PublishersController < AdminController
   def index
     @publishers = Publisher
     if params[:q].present?
-      @publishers = @publishers.where("email ILIKE :email OR name ILIKE :name",
-                      {
-                        email: "%#{params[:q]}%",
-                        name: "%#{params[:q]}%"
-                      }
-                    )
+      # Returns an ActiveRecord::Relation of publishers for pagination
+      @publishers = Publisher.where("publishers.id IN (#{sql(params[:q])})").distinct
```

Comment:
Instead of doing Publisher.where("publishers.id IN (#{sql(params[:q])})") which is essentially `select publishers.* from publishers where publishers.id in <YOUR SQL STATEMENT>`, you can simplify by doing `ActiveRecord::Base.connection.execute("<YOUR SQL CODE>")`

Reply:
`ActiveRecord::Base.connection.execute("<YOUR SQL CODE>")` would return a `PG::Result`, instead of an `ActiveRecord::Relation`, and we wouldn't be able to paginate.

`@publishers = Publisher.where("publishers.id IN (#{sql(params[:q])})").distinct` creates a relation of publishers sourced from the list of publisher ids returned by the `sql` method and allows for pagination.

I'm not sure if there is a better way of doing this.  I opted not to use Arel since this was quicker , and I'm not convinced it's more efficient.  Let me know

Reply:
Ah okay that makes sense.